src/product/product.service.ts

```diff
@@ -19,21 +19,23 @@ export class ProductService {
       rating,
       sortField,
       sortDirection,
+      searchValue,
     } = query;
     const skip = (page - 1) * limit;
 
     const dbQuery: any = {
-      ...(categoryId && { categoryId }),
-      ...(brands?.length && { brand: { $in: brands } }),
+      ...(categoryId ? { categoryId } : {}),
+      ...(brands?.length ? { brand: { $in: brands } } : {}),
       ...(priceMin !== undefined || priceMax !== undefined
         ? {
             price: {
-              ...(priceMin && { $gte: priceMin }),
-              ...(priceMax && { $lte: priceMax }),
+              ...(priceMin !== undefined ? { $gte: priceMin } : {}),
```

Comment:
Why `!== undefined` ?

Reply:
@VYaniv, because we have the case when we don't have priceMin or priceMax and this value may be converted to false. 
We can convert the value to boolean instead of  `!== undefined`